### PR TITLE
Remove learner member from endpointsFromCluster

### LINKED ISSUE #26: etcd cluster with a learner is not supported
Currently, we tried to use etcd-defrag to implement defragmentations on our etcd clusters, and we found it failed quickly due to that the **learner** node in cluster did not support health check.

Here is the execution log:
> Validating configuration.Validating the defragmentation rule: dbQuotaUsage > 0.8 || dbSizeFree/dbQuotaUsage > 0.5 ... validPerforming health check.{"level":"warn","ts":"2023-10-12T17:51:18.358902+0800","logger":"client","caller":"v3@v3.6.0-alpha.0.0.20230803155134-cca200345ab2/retry_interceptor.go:65","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc00030a000/[11.11.11.11:2379](http://11.11.11.11:2379/)","method":"/etcdserverpb.KV/Range","attempt":0,"error":"rpc error: code = **Unavailable desc = etcdserver: rpc not supported for learner"}**endpoint: https://11.11.11.11:2379/, health: false, took: 7.499546ms, error: etcdserver: rpc not supported for learnerendpoint: https://33.33.33.33:2379/, health: true, took: 7.733879ms, error:endpoint: https://44.44.44.44:2379/, health: true, took: 9.555876ms, error:endpoint: https://55.55.55.55:2379/, health: true, took: 10.164246ms, error:endpoint: https://66.66.66.66:2379/, health: true, took: 9.741549ms, error:endpoint: https://22.22.22.22:2379/, health: true, took: 43.014812ms, error:

So is this an ongoing issue?

### PR DESCRIPTION
**Pull request by git-yww**
Fix #26

### CODE REVIEW

File: endpoints.go [hunk 1]
```diff
@@ -48,7 +48,10 @@ func endpointsFromCluster(gcfg globalConfig) ([]string, error) {
 
 	var eps []string
 	for _, m := range memberlistResp.Members {
-		eps = append(eps, m.ClientURLs...)
+		// learner member only serves Status and SerializableRead requests, just ignore it
+		if !m.IsLearner {
```
Comment (ahrtr) [inline endpoints.go:52]:
```suggestion
		if !m.GetIsLearner() {
```

**Comment by git-yww** [comment]
> a quick question, have you verified this PR in your test that you mentioned in [#26 (comment)](https://github.com/ahrtr/etcd-defrag/issues/26#issuecomment-1766151933)?

Yes，it did work.

**Comment by ahrtr** [comment]
please squash the commits into one.

**Comment by git-yww** [comment]
> please squash the commits into one.

resolved

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @git-yww

**Comment by ahrtr** [comment]
Can you please add an item under https://github.com/ahrtr/etcd-defrag#note something like below?

```
Please do not get learner members' endpoints included in `--endpoints`,  refer to discussion in https://github.com/ahrtr/etcd-defrag/issues/26.
```

**Comment by git-yww** [comment]
Absolutely. That's a good point.

**Comment by git-yww** [comment]
Due to that one etcd cluster only supports one learner member, so the item to add will be like: 
`Please do not get learner member endpoint included in `--endpoints`,  refer to discussion in https://github.com/ahrtr/etcd-defrag/issues/26.`

**Comment by ahrtr** [comment]
The default maxLearner is 1, but actually it's configurable via the flag `--experimental-max-learners`, so let's use the plural noun.

File: README.md [hunk 2]
```diff
@@ -231,4 +231,5 @@ $ DOCKER_BUILDKIT=1 docker build --build-arg ARCH=${ARCH} -t "etcd-defrag:${VERS
 Any contribution is welcome!
 
 ## Note
-Please ensure running etcd on a version >= 3.5.6, and read [Two possible data inconsistency issues in etcd](https://groups.google.com/g/etcd-dev/c/8S7u6NqW6C4) to get more details.
+- Please ensure running etcd on a version >= 3.5.6, and read [Two possible data inconsistency issues in etcd](https://groups.google.com/g/etcd-dev/c/8S7u6NqW6C4) to get more details.
+- Please do not get learner member endpoint included in --endpoints, refer to discussion in https://github.com/ahrtr/etcd-defrag/issues/26.
```
Comment (ahrtr) [inline README.md]:
```suggestion
- Please do not get learner member endpoint included in `--endpoints`, refer to discussion in https://github.com/ahrtr/etcd-defrag/issues/26.
```

File: README.md [hunk 3]
```diff
@@ -231,4 +231,5 @@ $ DOCKER_BUILDKIT=1 docker build --build-arg ARCH=${ARCH} -t "etcd-defrag:${VERS
 Any contribution is welcome!
 
 ## Note
-Please ensure running etcd on a version >= 3.5.6, and read [Two possible data inconsistency issues in etcd](https://groups.google.com/g/etcd-dev/c/8S7u6NqW6C4) to get more details.
+- Please ensure running etcd on a version >= 3.5.6, and read [Two possible data inconsistency issues in etcd](https://groups.google.com/g/etcd-dev/c/8S7u6NqW6C4) to get more details.
+- Please do not get learner members' endpoints included in --endpoints, refer to discussion in https://github.com/ahrtr/etcd-defrag/issues/26.
```
Comment (ahrtr) [inline README.md:235]:
```suggestion
- Please do not get learner members' endpoints included in `--endpoints`, refer to discussion in https://github.com/ahrtr/etcd-defrag/issues/26.
```

**Comment by ahrtr** [comment]
Thanks again! Nice work!

**Pull request merged**